Maple2.Server.Game/Manager/Items/InventoryManager.cs

```diff
@@ -154,20 +154,20 @@ public class InventoryManager {
                 return false;
             }
 
+            if (!CanAdd(add)) {
+                return false;
+            }
+
             // If we are adding an item without a Uid, it may need to be created in db.
             if (add.Uid == 0) {
                 // Slot MUST be -1 so we don't add directly to a slot.
                 add.Slot = -1;
-                int remainStack = items.GetStackResult(add);
```

Comment:
you shouldn't remove this. If the item can fully stack into existing slots we don't need to create in DB.